poetry/installation/executor.py

```diff
@@ -42,14 +43,12 @@ def __init__(self, env, pool, config, io, parallel=None):
             parallel = config.get("installer.parallel", True)
 
         if parallel:
-            # This should be directly handled by ThreadPoolExecutor
-            # however, on some systems the number of CPUs cannot be determined
-            # (it raises a NotImplementedError), so, in this case, we assume
-            # that the system only has one CPU.
-            try:
-                self._max_workers = os.cpu_count() + 4
-            except NotImplementedError:
-                self._max_workers = 5
+            self._max_workers = get_default_max_workers()
```

Comment:
This logic seems quite jumbled up.

I'd suggest something like this instead (more elegant constructs are of course welcomed):

```python
            self._max_workers = config.get("installer.max-workers", get_default_max_workers())
            self._max_workers = min(self._max_workers, max_workers)
```